homeassistant/components/binary_sensor/generic_hygrostat.py

```diff
@@ -0,0 +1,217 @@
+        """Update local humidity state from source sensor."""
+        sensor = self.hass.states.get(self.sensor_id)
+
+        if sensor is None:
+            raise ValueError("Unknown humidity sensor '{}'".format(
+                self.sensor_id))
+
+        if sensor.state == STATE_UNKNOWN:
+            raise ValueError("Humidity sensor '{}' has state '{}'".format(
+                self.sensor_id, STATE_UNKNOWN))
+
+        try:
+            self.sensor_humidity = int(sensor.state)
+            self.add_sample(self.sensor_humidity)
+        except ValueError:
+            raise ValueError(
+                "Unable to update humidity from sensor '{}' with value '{}'"
+                .format(self.sensor_id, sensor.state))
+
+    def add_sample(self, value):
+        """Add given humidity sample to sample shift register."""
+        self.samples.append(value)
+
+    def calc_delta(self):
+        """Calculate the humidity delta."""
+        return self.sensor_humidity - self.get_minimum()
+
+    def get_minimum(self):
+        """Return the lowest humidity sample."""
+        try:
+            return min(self.samples)
+        except ValueError:
+            return None
+
+    def set_dehumidification_target(self):
+        """Setting dehumidification target to min humidity sample + offset."""
+        self.target = min(self.samples) + self.target_offset
+
+    def reset_dehumidification_target(self):
+        """Unsetting dehumidification target."""
+        self.target = None
+
+    def set_state(self, state):
+        """Setting hygrostat sensor to given state."""
+        if self._state is not state:
+            self._state = state
+            self.schedule_update_ha_state()
+
+    def set_max_on_timer(self):
+        """Setting max on timer."""
+        self.max_on_timer = time.time() + self.max_on_time
+
+    def reset_max_on_timer(self):
+        """Unsetting max on timer."""
+        self.max_on_timer = None
+
+    def set_on(self):
+        """Setting hygrostat to on."""
+        self.set_state(STATE_ON)
+        self.set_dehumidification_target()
+        self.set_max_on_timer()
+
+    def set_off(self):
+        """Setting hygrostat to off."""
+        self.set_state(STATE_OFF)
+        self.reset_dehumidification_target()
+        self.reset_max_on_timer()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the attributes of the entity."""
+        return {
+            ATTR_SENSOR_ID: self.sensor_id,
+            ATTR_DELTA_TRIGGER: self.delta_trigger,
+            ATTR_TARGET_OFFSET: self.target_offset,
+            ATTR_MAX_ON_TIME: self.max_on_time,
```

Comment:
These are configuration parameters, their static and don't belong in the attributes

Reply:
Keep the attributes list as short as possible